SSL Synthetics tests (#4623)
DataDog/documentation: content/en/synthetics/api_tests/_index.md

---
title: API Tests
kind: documentation
beta: true
description: Simulate and monitor HTTP requests from specific locations
aliases:
  - /synthetics/uptime_check
further_reading:
- link: "https://www.datadoghq.com/blog/introducing-synthetic-monitoring/"
  tag: "Blog"
  text: "Introducing Datadog Synthetics"
- link: "synthetics/"
  tag: "Documentation"
  text: "Manage your checks"
- link: "synthetics/browser_tests"
  tag: "Documentation"
  text: "Configure a Browser Test"
---

## Overview

> API tests are HTTP requests (`GET`, `POST`, `PUT`, etc.) executed by Datadog to your web properties or application endpoints at configurable periodic intervals [from multiple locations around the world][1]. These checks verify that your applications are responding to requests, as well as that they meet any conditions you define—such as response time, HTTP status code, and header or body contents.

Reviewer: it is wrong. Maybe the rebase has messed up everything 🤔

> API tests are useful to help you monitor your API endpoints and alert you when they are failing or too slow. These checks verify that your applications are responding to requests, as well as that they meet any conditions you define—such as response time, HTTP status code, and header or body contents. They are executed periodically from several locations around the world, the full list can be retrieved [through the Datadog API][1].

Reviewer 2: ```suggestion
API tests are useful to help you monitor your API endpoints and alert you when they are failing or too slow. These checks verify that your applications are responding to requests, as well as that they meet any conditions you define—such as response time, HTTP status code, and header or body contents. Use the[Datadog API][1] to see the full list.
```

## Configuration

API tests configuration depends on the type of API test you want to create. There are two API test types: [HTTP test](?tab=httptest) and [SSL test](?tab=ssltest).

### Make a request

Define the `HTTP` or `SSL` request you want to be executed by Datadog:

{{< tabs >}}

{{% tab "HTTP Test" %}}

{{< img src="synthetics/api_tests/make-http-request.png" alt="Make HTTP Request" responsive="true" style="width:80%;" >}}

> Define the request you want executed by Datadog:

Reviewer 2: ```suggestion
Define the request you want to be executed by Datadog:
```

1. Select users and/or [services][3] to send the notifications to. Note that you can use the [`@-notification` feature][4] in the **message** field.
2. Enter a **message** for the API test. This field allows standard [Markdown formatting][5]. Notification messages include the **message** defined in this section and information about which assertion failed and why.
3. Click **Save** to save your API test.

Notifications example:

{{< img src="synthetics/api_tests/notifications-example.png" alt="API Test Notifications" responsive="true" style="width:80%;" >}}

## Network timings

The Synthetics details page displays the following network timings:

| Timing                    | Description                                                                                           |
|---------------------------|-------------------------------------------------------------------------------------------------------|
| `DNS`                       | Time spent resolving the DNS name of the last request.                                                |
| `Connect`                   | Time spent establishing a connection to the server.                                                   |

> | `SSL`                       | Time spent for the TLS handshake. If the last request is not over HTTPS, this metric does not appear. |

Reviewer: missing the link to the ssl errors.

Author: the link is above in the Test Failure section :)
